functions/supplier/sftp-supplier/src/main/java/org/springframework/cloud/fn/supplier/sftp/SftpSupplierConfiguration.java

```diff
@@ -221,6 +226,12 @@ public RemoteFileDeletingAdvice remoteFileDeletingAdvice(SftpRemoteFileTemplate
 			return new RemoteFileDeletingAdvice(sftpTemplate, sftpSupplierProperties.getRemoteFileSeparator());
 		}
 
+		@Bean
+		@ConditionalOnExpression("environment['sftp.supplier.rename-remote-files-to'] != null && environment['sftp.supplier.delete-remote-files'] == 'false'")
```

Comment:
We don't need so complex condition: you cover mutual exclusion in the configuration properties validation.
So, this one probably could be just `@ConditionalOnProperty(prefix = "sftp.supplier", value = "rename-remote-file-to")`